driver-core/src/main/java/com/datastax/driver/core/policies/ErrorAwarePolicy.java

```diff
@@ -0,0 +1,322 @@
+    }
+
+    @Override
+    public Iterator<Host> newQueryPlan(String loggedKeyspace, Statement statement) {
+        final Iterator<Host> childQueryPlan = childPolicy.newQueryPlan(loggedKeyspace, statement);
+
+        return new AbstractIterator<Host>() {
+
+            @Override
+            protected Host computeNext() {
+                long currentTime = System.currentTimeMillis();
+                while (childQueryPlan.hasNext()) {
+                    Host host = childQueryPlan.next();
+
+                    Long excludedTime = errorTracker.getExclusions().get(host);
+                    if (excludedTime == null) {
+                        // No entry in exclusions map means host hasn't crossed the threshold yet.
+                        return host;
+                    }
+
+                    if (currentTime - excludedTime > retryPeriodMillis) {
+                        return host;
+                    }
+                }
+
+                return endOfData();
+            }
+        };
+    }
+
+    @Override
+    public void onAdd(Host host) {
+        childPolicy.onAdd(host);
+    }
+
+    @Override
+    public void onUp(Host host) {
+        childPolicy.onUp(host);
+    }
+
+    @Override
+    public void onDown(Host host) {
+        childPolicy.onDown(host);
+    }
+
+    @Override
+    public void onRemove(Host host) {
+        childPolicy.onRemove(host);
+    }
+
+    /**
+     * Creates a new error aware policy builder given the child policy
+     * that the resulting policy should wrap.
+     *
+     * @param childPolicy the load balancing policy to wrap with error
+     *                    awareness.
+     * @return the created builder.
+     */
+    public static Builder builder(LoadBalancingPolicy childPolicy) {
+        return new Builder(childPolicy);
+    }
+
+    @Override
+    public void close() {
+        childPolicy.close();
+    }
+
+    /**
+     * Utility class to create a {@link ErrorAwarePolicy}.
+     */
+    public static class Builder {
+        final LoadBalancingPolicy childPolicy;
+
+        double errorsThreshold = 1.0;
+        int retryPeriod = 5000;
+
+        ErrorsFilter errorsFilter = new DefaultErrorsFilter();
+
+        /**
+         * Creates a {@link Builder} instance.
+         *
+         * @param childPolicy the load balancing policy to wrap with error
+         *                    awareness.
+         */
+        public Builder(LoadBalancingPolicy childPolicy) {
+            this.childPolicy = childPolicy;
+        }
+
+        /**
+         * Defines the limit of errors after which a host will be considered INVALID
+         * by the {@link ErrorAwarePolicy}.
+         * <p/>
+         * The error tracker is going to get notified right away, by the {@link Cluster} it is registered to, of each error
+         * happening for each hosts.
+         * <p/>
+         * Internally a {@link Meter} is kept up to date for each host upon each error. The Error aware policy then,
```

Comment:
the comma in The Error aware policy then*,*  is extraneous and can be removed i think.